sycl/plugins/unified_runtime/CMakeLists.txt

```diff
@@ -63,20 +58,39 @@ add_sycl_plugin(unified_runtime
     "pi2ur.cpp"
     "ur/ur.hpp"
     "ur/ur.cpp"
-    "ur/adapters/level_zero/ur_level_zero.hpp"
-    "ur/adapters/level_zero/ur_level_zero.cpp"
     "ur/usm_allocator.hpp"
     "ur/usm_allocator.cpp"
     # These below belong to Unified Runtime PI Plugin only
     "pi_unified_runtime.hpp"
     "pi_unified_runtime.cpp"
     LIBRARIES
     Threads::Threads
+    UnifiedRuntimeLoader
+    UnifiedRuntime-Headers
+)
+
+# Build level zero adapter
+add_sycl_library("ur_adapter_level_zero" SHARED
+  LINKER_SCRIPT "${PROJECT_SOURCE_DIR}/plugins/ld-version-script.txt"
+  SOURCES
+    "ur/ur.hpp"
+    "ur/ur.cpp"
+    "ur/adapters/level_zero/ur_level_zero.hpp"
+    "ur/adapters/level_zero/ur_level_zero.cpp"
+  INCLUDE_DIRS
+  LIBRARIES
     UnifiedRuntime-Headers
+    UnifiedRuntimeLoader
```

Comment:
Why do you need to link with the loader here?

Reply:
good catch I don't think it is needed, i will remove